galgebra/mv.py

```diff
@@ -1468,23 +1473,24 @@ def consolidate_coefs(sdop):
                 else:
                     new_coefs.append(coef)
                     new_pdiffs.append(pd)
-        new_terms = zip(new_coefs, new_pdiffs)
+        new_terms = list(zip(new_coefs, new_pdiffs))
 
         if isinstance(sdop, Sdop):
             return Sdop(new_terms, ga=sdop.Ga)
         else:
             return new_terms
 
     def simplify(self, modes=simplify):
-        coefs, pdiffs = zip(*self.terms)
+        coefs, pdiffs = list(zip(*self.terms))
         new_coefs = []
         for coef in coefs:
             new_coefs.append(metric.apply_function_list(modes,coef))
-        self.terms = zip(new_coefs,pdiffs)
+        self.terms = list(zip(new_coefs,pdiffs))
         return self
 
     def sort_terms(self):
-        self.terms.sort(key=operator.itemgetter(1), cmp=Pdop.compare)
+        # self.terms.sort(key=operator.itemgetter(1))
+        self.terms.sort(key=cmp_to_key(Pdop.compare))
```

Comment:
It was using both key and cmp, and the cmp obvious contains more logics which I chose to keep instead of key.